addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -314,6 +314,10 @@ class MessageList extends Component {
      * @param {Object} hint
      */
     async _adjustFromChangeOfThreadCache(hint) {
+        if (!this.props.hasScrollAdjust) {
+            this.threadViewer.markComponentHintProcessed(hint);
+            return;
+        }
```

Comment:
I expect conditional with `hasScrollAdjust` props around scroll actions in adjustment methods.
Doing a preemptive return at the top of each of these methods only work if all adjustments always mark hint as processed and the actions in these methods are only scrolls.

Not all adjustments do necessarily mark the hint as processed (e.g. message to scroll is not visible in UI yet, due to async rendering of OWL), and component hints have been introduced to solve scrolling but they have not been designed just for scrolling (hence why you don't see the word "scroll" in these adjust methods, nor in the content of hints).

Keep in mind hints are temporary logs of dynamic changes in models that can be useful for components.
The word "adjustment" just means a response to a hint, so by the name of your prop, this should dictate whether scroll actions are performed inside adjustment methods. It's wrong for it to just flag hint as processed and do nothing (what if some adjustment actions were not just scrolls, like focus?).

Reply:
@alexkuhn 
I am not sure I understand this.
I agree with you on the purpose of hints, no problem
But for message_list we use them only for scrolling
the caller function dispatches towards the right method, and again, currently only those scrolling methods are called, each in one case.
The default behavior of the caller function is to mark as processed

I think what you propose may be correct in the future, but not yet, and not even as projects
In short, I understood your comment as more engineering of what to ignore and how, which sounds like over engineering to me
Isn't it ?

Reply:
You think it sounds like over-engineering because the diff is bit bigger with this change, and also because, right now, both solutions emit same functional behaviour.

But if you really think about it, the shorter diff increases maintenance cost quite substantially: you have a prop whose naming is misleading with the way it's managed, and the pre-emptive return code bear little context to why this make sense.

Maybe you might think I making a fuss for such a little stuff, but just imagine few months from now you have to read this code and understand why you did that... Also imagine you applied the same "laziness" (I meant the code, not in person! I just couldn't find a better word...) several times, and there are lots of strange and unexpected interactions between these "over-simplifications" of logic.